gprofiler/profilers/profiler_base.py

```diff
@@ -171,6 +171,11 @@ def _profiling_error_stack(
     def snapshot(self) -> ProcessToProfileData:
         processes_to_profile = self._select_processes_to_profile()
         logger.debug(f"{self.__class__.__name__}: selected {len(processes_to_profile)} processes to profile")
+        if self._profiler_state.processes_to_profile is not None:
+            processes_to_profile = [
+                process for process in processes_to_profile if process in self._profiler_state.processes_to_profile
+            ]
+            logger.debug(f"{self.__class__.__name__}: processes left after filtering: {len(processes_to_profile)}")
```

Comment:
UX wise - no need to log this line if `len(processes_to_profile) == 0` to begin with, let's skip this `if` is that's the case?